.github/coding_guidelines/CODING_GUIDELINES.md

```diff
@@ -0,0 +1,76 @@
+# MetaMask Coding Guidelines
+
+### 1. New Code Should be Typescript
+- New components and utilities should be written in Typescript and enforce typing.
+- Existing code should be refactored into Typescript where time allows. If you are replacing a component use Typescript.
+
+### 2. Using Functional Components and Hooks Instead of Classes
+- Use functional components and hooks as they result in more concise and readable code compared to classes.
+
+### 3. Organize Files Related to the Same Component in One Folder
+- An example of a component file structure:
+
+```.tsx
+AvatarAccount
+├── AvatarAccount.constants.ts
+├── AvatarAccount.stories.tsx
+├── AvatarAccount.styles.ts
```

Comment:
we are still using scss for styles, i would say this can be `avatar-account.scss`?

Reply:
I'm wondering what is the long term guideline we want to promote for styles? `scss` or `ts`?
In Mobile `ts` is indicated in the [guideline](https://github.com/MetaMask/metamask-mobile/blob/3f43e586d2e442b66295d1ede01eec49ead91146/.github/coding_guidelines/CODING_GUIDELINES.md?plain=1#L17).

Is there a reason to have a different guideline for Extension and Mobile?

Reply:
Actually i don't have a concrete answer for this, i would say @georgewrmarshall might have better idea about this one !

Reply:
Hi @NicolasMassart , as I'd like us to have the same coding guidelines on Extension and Mobile side, can I please ask for your recommendation on this question?

Reply:
Hi @georgewrmarshall @NicolasMassart do you have any recommendation on this topic?

Reply:
Hey @gauthierpetetin @DDDDDanica, Apologies for the late reply. We use `.ts` in mobile and `.scss` in the extension codebases for distinct reasons relating to how styles are written in different languages.

- In mobile, we are using a CSS-in-JS approach where the styles are written in TypeScript. 
- In extension, we write styles in SCSS which is transpiled into CSS.

We can't consolidate the 2 here.

Reply:
Ok makes sense, thanks for the detail. It now updated in this [commit](https://github.com/MetaMask/metamask-extension/pull/20651/commits/4b5fdc9c3ee5acfc767c7ca5668badec31c58cb9).